aggregate_temporal_frequency.json

```diff
@@ -0,0 +1,112 @@
+            "schema": {
+                "type": "object",
+                "subtype": "process-graph",
+                "parameters": [
+                    {
+                        "name": "data",
+                        "description": "A labeled array with elements of any type.",
+                        "schema": {
+                            "type": "array",
+                            "subtype": "labeled-array",
+                            "items": {
+                                "description": "Any data type."
+                            }
+                        }
+                    },
+                    {
+                        "name": "context",
+                        "description": "Additional data passed by the user.",
+                        "schema": {
+                            "description": "Any data type."
+                        },
+                        "optional": true,
+                        "default": null
+                    }
+                ]
+            }
+        },
+        {
+            "name": "dimension",
+            "description": "The name of the temporal dimension for aggregation. All data along the dimension will be passed through the specified reducer. If the dimension is not set or set to `null`, the data cube is expected to only have one temporal dimension. Fails with a `TooManyDimensions` error if it has more dimensions. Fails with a `DimensionNotAvailable` error if the specified dimension does not exist.\n\n**Note:** The default dimensions a data cube provides are described in the collection's metadata field `cube:dimensions`.",
+            "schema": {
+                "type": [
+                    "string",
+                    "null"
+                ]
+            },
+            "default": null,
+            "optional": true
+        },
+        {
+            "name": "context",
+            "description": "Additional data to be passed to the reducer.",
+            "schema": {
+                "description": "Any data type."
+            },
+            "optional": true,
+            "default": null
+        }
+    ],
+    "returns": {
+        "description": "A data cube with potentially lower resolution and cardinality, but the same number of dimensions as the original data cube. The specified temporal dimension has the following dimension labels (`YYYY` = four-digit year, `MM` = two-digit month, `DD` two-digit day of month):\n\n* `hourly`: `YYYY-MM-DD-0` - `YYYY-MM-DD-23`\n* `daily`: `YYYY-1` - `YYYY-365`\n* `weekly`: `YYYY-1` - `YYYY-52`\n* `monthly`: `YYYY-1` - `YYYY-12`\n* `yearly`: `YYYY`\n* `seasons`: `YYYY-djf` (December - February), `YYYY-mam` (March - May), `YYYY-jja` (June - August), `YYYY-son` (September - November).\n* `tropical_seasons`: `YYYY-ndjfma` (November - April), `YYYY-mjjaso` (May - October).",
```

Comment:
maybe more consistent to always same number of digits for hours (`00`-`23`), days (`001`-`365`), ...

Reply:
That comes from aligning with climatological_normal, where we store integer labels and thus 001 would be simply 1...

Reply:
I meant these, where offsets are converted to string representation:

> `hourly`: `YYYY-MM-DD-0` - `YYYY-MM-DD-23`
> `daily`: `YYYY-1` - `YYYY-365`
> `weekly`: `YYYY-1` - `YYYY-52`
> `monthly`: `YYYY-1` - `YYYY-12`

mismatch between alphabetical and logical sorting will  cause headaches somewhere down the line

Reply:
Yes, I understood that. Still, for consistentcy we'd should then pad the numbers in the other processes, too.

Reply:
> we'd should then pad the numbers in the other processes, too

certainly, I added this comment to this line, but it's also about all other places where this pattern occurs of concatenating ints in string.

Also, I assume the `YYYY`, `MM` and `DD` are also zero-padded, so I would just do it for all components of these strings (where possible)

Reply:
Changed this in the last commit, please check. I'll check the implementation for potential issues.